gadget/install/install.go

```diff
@@ -251,11 +256,23 @@ func ensureLayoutCompatibility(gadgetLayout *gadget.LaidOutVolume, diskLayout *g
 		return false, reasonAbsent
 	}
 
+	// check size of volumes
 	if gadgetLayout.Size > diskLayout.Size {
 		return fmt.Errorf("device %v (%s) is too small to fit the requested layout (%s)", diskLayout.Device,
 			diskLayout.Size.IECString(), gadgetLayout.Size.IECString())
 	}
 
+	// check that the sizes of all structures in the gadget are multiples of
+	// the disk sector size (unless the structure is the MBR)
+	for _, ls := range gadgetLayout.LaidOutStructure {
+		if !gadget.IsRoleMBR(ls) {
```

Comment:
I think you could use `gadget.IsPartition()` here, which handles MBR internally too.

Reply:
Hmm, so I couldn't find any examples of `type: bare` gadget.yaml's, so it's probably not a bad idea, but this does mean that any bare structures on the disk are not verified to be in alignment with the sector size, not sure if we should call that out somewhere. 

However, I think it's usually safer to go from more strict -> less strict, as going the other way we have had issues with, so I think I'd rather stick with this PR as being more strict and doing verification on bare structures too, even if it is awkward to use this IsRoleMBR function.

Reply:
Ok, let's keep it like this. The previous code in layout.go behaved similarly, so it's probably ok to not change the behavior.